tablo/models.py

```diff
@@ -700,13 +708,23 @@ def add_feature(self, feature):
         )
 
         date_fields = [field['name'] for field in self.fields if field['type'] == 'esriFieldTypeDate']
+        image_fields = [field['name'] for field in self.fields if field['type'] == 'esriFieldTypeBlob']
+        images_large = {}
+        images_thumbs = {}
         values = []
         for attribute_name in colnames_in_table:
             if attribute_name in date_fields and feature['attributes'][attribute_name]:
                 if isinstance(feature['attributes'][attribute_name], str):
                     values.append(feature['attributes'][attribute_name])
                 else:
                     values.append(datetime.fromtimestamp(feature['attributes'][attribute_name] / 1000))
+
+            elif attribute_name in image_fields and feature['attributes'][attribute_name]:
+                image_path = FeatureServiceLayer.create_image_path(self.service.id, "NO_PRIMARY_KEY", attribute_name)
```

Comment:
Its a placeholder. We don't have the PK yet. I create a dict entry where the key is the path with a placeholder key, which is later replaced with the real PK. The value is the actual image, which I need to create anyway to create the thumbnail.

I realize this is a little confusing, but it avoids creating the image twice, which is expensive. 

The alternative would be to do the insert without the thumbnail, then loop through all records and create the image, upload to S3, and insert the thumbnail. Probably clearer, but two inserts per record.